### PR TITLE
Add segment reg support and numerous fixes FIX #5967

### LINKED ISSUE #5967: X86 segment override confuses rasm2
Example program for gcc: `int main() { __asm__ ( "movl %fs:(1), %eax\n"); }`

r2's `pd 10 @main` prints `0x004004da      648b04250100.  mov eax, dword fs:[1]`

`rasm2 -D 648b042501000000` prints the same. 

However it can't convert it back.
`rasm2 'mov eax, dword fs:[1]'` = 89e0 (mov eax, esp)

Also if you start to play with syntax, rasm2 starts misbehave even more:

`rasm2 -b 32 'mov edi, fs:[65534]'` = 89e7 (sameish)
`rasm2 -b 32 'mov edi, [fs:65534]'` = 8a3ce500000000 (even more strange `mov bh,BYTE PTR [eiz*8+0x0]`)

(output is same for -b32 or -b64)

It also doesn't report error on something silly like 

`rasm2 'mov fs:[1],ds:[1]'` (88dc)
`rasm2 'mov fs:[1],[ds:1]'` (8a23)
`rasm2 'mov [fs:1],ds:[1]'` (881c24)
`rasm2 'mov [fs:1],[ds:1]'` (8a03)

### PR DESCRIPTION
**Pull request by Svenito**
Add support for segment registers with offset for `mov` command, and segment reg support for `pop` and `push` ops

Includes a number of fixes for 64bits ops too.

r2r pr accompanies this commit

### CODE REVIEW

File: libr/asm/p/asm_x86_nz.c [hunk 1]
```diff
@@ -1963,6 +2119,10 @@ static int parseOperand(RAsm *a, const char *str, Operand *op) {
 			op->extended = true;
 			op->reg -= 8;
 		}
+		if (op->type & OT_REGTYPE & OT_SEGMENTREG) {
+			parse_segment_offset(a, str, &nextpos, op, reg_index);
```
Comment (radare) [inline libr/asm/p/asm_x86_nz.c]: missing space before (

File: libr/asm/p/asm_x86_nz.c [hunk 2]
```diff
@@ -1761,7 +1890,7 @@ static Register parseReg(RAsm *a, const char *str, size_t *pos, ut32 *type) {
 		// This isn't working properly yet
 		for (i = 0; sregs[i]; i++) {
 			if (!strncasecmp (sregs[i], token, length)) {
-				*type = (OT_GPREG & OT_REG(i)) | OT_BYTE;
+				*type = (OT_SEGMENTREG & OT_REG(i)) | OT_WORD;
```
Comment (radare) [inline libr/asm/p/asm_x86_nz.c]: missing space before (

**Comment by radare** [comment]
Please rebase, travis output is a bit wtf now

**Comment by radare** [comment]
screen shot 2016-10-19 at 13 05 00

**Comment by Svenito** [comment]
rebased and whitespace fixed

**Pull request merged**